### PR TITLE
fix(slots): consume eagerly rendered slot after one use

### LINKED ISSUE #7068: Different behavior with slot vs Astro.slots.render
### What version of `astro` are you using?

2.4.4

### Are you using an SSR adapter? If so, which one?

node

### What package manager are you using?

npm

### What operating system are you using?

Mac

### What browser are you using?

Firefox

### Describe the Bug

When using the `<slot/>` syntax and rendering the same component twice, the slot content will be empty.
Not happening when doing with `Astro.slots.render`

### Link to Minimal Reproducible Example

https://codesandbox.io/p/sandbox/priceless-solomon-sxv6r4?on=codesandbox&file=%2Fsrc%2Fcomponents%2FCardContainer.astro%3A1%2C1

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by lilnasy**
## Changes
- Fixes #7068 

## Testing
Added a case to an existing fixture.

## Docs
Does not affect usage.

### CODE REVIEW

File: .changeset/plenty-spiders-act.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes an issue where a slot rendered multiple times resulted in the same output each time.
```
Comment (sarah11918) [inline .changeset/plenty-spiders-act.md:5]:
```suggestion
Fixes an issue where rendering multiple slots resulted in the same output each time.
```
super-nit: "a slot" rendered multiple times makes me think it's the same one slot, so the output should be the same! Would this be a correct description? (non-blocking, though)

Comment (lilnasy) [reply .changeset/plenty-spiders-act.md:5]: It is the same one slot.

Comment (lilnasy) [reply .changeset/plenty-spiders-act.md:5]: Now that you bring it up though, I think there is room for clarification. Before this PR, the first render was reused when users tried to manually render the same slot multiple times.

File: packages/astro/src/runtime/server/render/astro/instance.ts [hunk 2]
```diff
@@ -31,8 +31,18 @@ export class AstroComponentInstance {
...
+			let value: ReturnType<typeof slots[string]> | null = slots[name](result);
+			this.slotValues[name] = () => {
+				// use prerendered value only once
+				if (value !== null) {
+					const consumedValue = value
+					value = null
+					return consumedValue
+				}
+				// render afresh for the advanced use-case where the same slot is rendered multiple times 
+				return slots[name](result);
```
Comment (natemoo-re) [inline packages/astro/src/runtime/server/render/astro/instance.ts:44]: This code definitely works, but I feel like a simple toggle might be easier to understand? Something like...

```js
let didRender = false;
let value = slots[name](result);
this.slotValues[name] = () => {
  if (!didRender) {
		didRender = true;
    return value;
  }
  return slots[name](result);
```

**Review by natemoo-re — APPROVED** [review]
Implementation LGTM! I think it could be refactored to be slightly more clear, but it's up to you. I wouldn't block a merge as-is.

File: packages/astro/src/runtime/server/render/astro/instance.ts (hunk 2 above)
Comment (lilnasy) [reply packages/astro/src/runtime/server/render/astro/instance.ts]: Much cleaner, thanks!

**Review by matthewp — APPROVED** [review]

File: .changeset/plenty-spiders-act.md (hunk 1 above)
Comment (sarah11918) [reply .changeset/plenty-spiders-act.md:5]: OK, then maybe something like.... (wordy, but trying to see if this is the right idea)

> Fixes an issue where a slot rendered multiple times incorrectly used the same initial rendering for each instance

**Review by sarah11918 — APPROVED** [review]
LGTM!

**Pull request merged**